k2/csrc/dtype.h

```diff
@@ -43,16 +43,16 @@ class DtypeTraits {
 
  private:
   // We may add more
-  char base_type_;    // BaseType converted to char
-  char num_scalars_;  // currently always 1; may be greater for vector types in
+  BaseType base_type_;    // BaseType converted to char
+  int32_t num_scalars_;  // currently always 1; may be greater for vector types in
                       // future.  Must divide num_bytes exactly.
-  char misc_;  // field that is normally 0, but we may use for extensions in
+  int32_t misc_;  // field that is normally 0, but we may use for extensions in
                // future.
-  char num_bytes_;  // sizeof() this type in bytes, gives stride.  The size per
+  int32_t num_bytes_;  // sizeof() this type in bytes, gives stride.  The size per
```

Comment:
I guess @danpovey use `char` here (for `misc_`, `num_bytes_`, etc.) intentionally as we don't need 4 bytes to store those information

Reply:
Just wonder, is `int8_t` a good replacement for these. Need @danpovey help here.

Reply:
yes, int8_t would be good.  I think my idea was it may be nice to load as a single word from memory.